pkg/theia/commands/anomaly_detection_algos.go

```diff
@@ -0,0 +1,226 @@
+// Copyright 2022 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package commands
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+	"strings"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/spf13/cobra"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	anomalydetector "antrea.io/theia/pkg/apis/anomalydetector/v1alpha1"
+	"antrea.io/theia/pkg/theia/commands/config"
+	"antrea.io/theia/pkg/util"
+)
+
+// throughputAnomalyDetectionEWMACmd represents the policy-recommendation delete command
+var throughputAnomalyDetectionAlgoCmd = &cobra.Command{
+	Use:   "run",
+	Short: "throughput Anomaly detection using Algo",
+	Long:  `throughput Anomaly detection using Algorithms, currently supported Algorithms are EWMA, ARIMA and DBSCAN`,
+	Example: `Run the spefic Algorithm for throughput Anomaly Detection
+	$ theia throughput-anomaly-detection run --algo ARIMA --start-time 2022-01-01T00:00:00 --end-time 2022-01-31T23:59:59
+	Run thoughput anomaly detection algorithm of type ARIMA starting from '2022-01-01 00:00:00' until '2022-01-31 23:59:59'`,
+	RunE: throughputAnomalyDetectionAlgo,
+}
+
+var (
+	algoName string
+)
+
+func throughputAnomalyDetectionAlgo(cmd *cobra.Command, args []string) error {
+	throughputAnomalyDetection := anomalydetector.ThroughputAnomalyDetector{}
+	algoName, err := cmd.Flags().GetString("algo")
+	if err != nil {
+		return err
+	}
+	if algoName == "" && len(args) == 1 {
+		algoName = args[0]
+	}
+	err = util.ParseADAlgorithmName(algoName)
+	if err != nil {
+		return err
+	}
+	throughputAnomalyDetection.Type = algoName
+
+	startTime, err := cmd.Flags().GetString("start-time")
+	if err != nil {
+		return err
+	}
+	startTime = strings.Replace(startTime, "T", " ", 1)
```

Comment:
As you need to replace the `T` here, maybe just let user to provide a time in the format of `YYYY-MM-DD hh:mm:ss`? And I think this is also the format we used in policy recommendation.

Reply:
I have observed with my Mac, that when we provide the argument as --start-time “YYYY-MM-DD hh:mm:ss”, the os is parsing start time as "YYYY-MM-DD separating the argument with space, thus to keep it simple I used the spaceless format.

Reply:
I think to allow a space in the argument, you only need to wrap the argument with quotes like what we do in the policy recommendation case
`theia policy-recommendation run --type initial --policy-type anp-deny-applied --start-time '2022-01-01 00:00:00' --end-time '2022-01-31 23:59:59'`

Reply:
yes, I tried doing that, and have also seen it working with quotes but with my Mac it is not working, in order to later not face this issue, I changed the format to spaceless

Reply:
That's strange. May I check if you are using “ instead of " or ' for the quote?

Reply:
I have tried both ‘ and ` in [throughputanomalydetector_test.go](https://github.com/antrea-io/theia/pull/152/files#diff-ca2928dc4a5a98ec0665e5f231b4c926099b263b51ec1ec8e771f55a81916d72) and both have failed, I can not use “ as that would end the string 

stderr:Error: parsing start-time: parsing time "`2022-08-11" as "2006-01-02 15:04:05": cannot parse "`2022-08-11" as "2006", start-time should be in
                                'YYYY-MM-DDThh:mm:ss' format, for example: 2006-01-02T15:04:05

Reply:
I might not express correctly. I mean the correct one should be " or '. As I see you using --start-time “YYYY-MM-DD hh:mm:ss” in the previous comment, I would like to verify whether there is a typo in your tries. If you finally fails to use the one with the space, let's keep the spaceless version. But instead of replacing the `T`, you can try `2006-01-02T15:04:05` for time parsing layout.

Reply:
with the current code, user can provide both spaceless and spaced version.

Reply:
I see. Let's keep it as it is before figuring out why the version with space does not work.